### PR TITLE
Allow the option labels of NullableBooleanInput to be customized

### PR DESCRIPTION
**Pull request by gavacho**
Hi, I would like to customize the labels of the options of the `NullableBooleanInput` control.

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx [hunk 1]
```diff
@@ -108,12 +101,12 @@ const NullableBooleanInput: FunctionComponent<
...
-            {...enhancedOptions}
+            {...options}
             {...sanitizeRestProps(rest)}
         >
-            <MenuItem value="">{translate('ra.boolean.null')}</MenuItem>
-            <MenuItem value="false">{translate('ra.boolean.false')}</MenuItem>
-            <MenuItem value="true">{translate('ra.boolean.true')}</MenuItem>
+            <MenuItem value="">{translate(nullLabel)}</MenuItem>
+            <MenuItem value="false">{translate(falseLabel)}</MenuItem>
+            <MenuItem value="true">{translate(trueLabel)}</MenuItem>
```
Comment (gavacho) [inline packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx:109]: Should I keep these call to `translate` here? Or should consumers of this component be expected to translate their own values?

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx [hunk 2]
```diff
@@ -70,20 +77,6 @@ const NullableBooleanInput: FunctionComponent<
         validate,
     });
 
-    const enhancedOptions = displayNull
-        ? {
-              ...options,
-              SelectProps: {
-                  displayEmpty: true,
```
Comment (gavacho) [inline packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx:77]: The property is named `allowEmpty` so this has no effect. This component is already adding a "null" option.

**Comment by fzaninotto** [comment]
Why don't you want to use the current solution (using translation messages)?

**Comment by gavacho** [comment]
@fzaninotto I have two different NullableBooleanInputs in my app and want to use different labels on them.

**Comment by gavacho** [comment]
@fzaninotto The default behavior of this control is to display 3 items, the item representing the null item has a blank label so the options, while present in the dropdown, is very difficult to select.

This has created confusion in the past. For example, this PR https://github.com/marmelab/react-admin/pull/4365 is addressing a problem that doesn't exist (because the null option is always rendered by NullableBooleanInput).

Maybe NullableBooleanInput should guard against the case of the translation message being an empty string.

**Comment by fzaninotto** [comment]
PR #5326 should solve the bug of the null option not being visible enough.

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx (hunk 1 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx:109]: you should keep the calls to `translate`

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx [hunk 3]
```diff
@@ -30,7 +30,12 @@ const getStringFromBoolean = (value?: boolean | null): string => {
 };
 
 const NullableBooleanInput: FunctionComponent<
-    InputProps<TextFieldProps> & Omit<TextFieldProps, 'label' | 'helperText'>
+    InputProps<TextFieldProps> &
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx:33]: this type is becoming increasingly complex. Anyone trying to extend `NullableBooleanInput` in TypeScript will fight trying to type its props. 

Please create a custom `NullableBooleanInputProps` interface and export it.

File: docs/Inputs.md [hunk 4]
```diff
@@ -450,30 +450,20 @@ import { NullableBooleanInput } from 'react-admin';
...
 import { NullableBooleanInput } from 'react-admin';
 
-<NullableBooleanInput 
+<NullableBooleanInput
     label="Commentable"
     source="commentable"
-    displayNull
+    nullLabel="Either"
+    falseLabel="No"
+    trueLabel="Yes"
```
Comment (fzaninotto) [inline docs/Inputs.md:463]: You should mention that these labels are translated by react-admin.

File: docs/Inputs.md [hunk 5]
```diff
@@ -450,30 +450,20 @@ import { NullableBooleanInput } from 'react-admin';
 
 ![NullableBooleanInput](./img/nullable-boolean-input.png)
 
-`<NullableBooleanInput />` doesn't display the empty option by default. If you want to customize its label and display it, you can use the `displayNull` prop.
+The labels of the options can be customized using the `nullLabel`, `falseLabel` and `trueLabel` properties.
```
Comment (fzaninotto) [inline docs/Inputs.md:453]: This is a bit confusing: the doc should explain that in order to customize the labels for the entire app, the right way is to override the translation. Custom labels are only necessary if one wants to have distinct labels for one given input.

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.spec.tsx [hunk 6]
```diff
@@ -138,4 +138,70 @@ describe('<NullableBooleanInput />', () => {
             'true'
         );
     });
+
+    it('should allow to customize the label of the null option', () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/NullableBooleanInput.spec.tsx:142]: 👍 Excellent, thanks for thinking of adding tests!

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Comment by gavacho** [comment]
Oops, I just realized this was opened against the `master` branch upstream. Shall I re-open against `next`?

**Comment by djhi** [comment]
> Oops, I just realized this was opened against the master branch upstream. Shall I re-open against next?

The next release will be 3.9 and we already moved the related code to master so you're fine :)

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Perfect, thanks a lot!

**Pull request merged**